src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/addhearingoutcome/AddHearingOutcomeAboutToStartHandler.java

```diff
@@ -56,17 +61,33 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
             HearingsGetResponse response = hmcHearingsApiService.getHearingsRequest(Long.toString(caseDetails.getId()), HmcStatus.COMPLETED);
             List<CaseHearing> hmcHearings = response.getCaseHearings();
             if (!hmcHearings.isEmpty()) {
-                List<HearingDetails> selectedHearings = sscsCaseData.getHearings()
-                        .stream()
-                        .map(Hearing::getValue)
-                        .filter(value -> value.getStart() != null)
-                        .filter(value -> hmcHearings.stream()
-                                .anyMatch(hmcHearing -> Objects
-                                        .equals(hmcHearing.getHearingId().toString(), value.getHearingId())))
-                        .sorted(Comparator.comparing(HearingDetails::getStart).reversed())
-                        .toList();
+                Map<String, CaseHearing> hmcHearingsMap = hmcHearings.stream()
+                    .collect(Collectors.toMap(caseHearing -> caseHearing.getHearingId().toString(),
+                        caseHearing -> caseHearing));
+
+                if (sscsCaseData.getCompletedHearingsList() == null) {
+                    sscsCaseData.setCompletedHearingsList(new ArrayList<>());
+                }
+
+                sscsCaseData.getCompletedHearingsList().addAll(sscsCaseData.getHearings()
```

Comment:
I think since lot is happening here I would suggest we add some logs like how many records where fetched and hearing id epics id and start/end date for each that would be good to debug if we need to

Reply:
Logging added